src/main.rs

```diff
@@ -62,3 +67,7 @@ pub fn main() {
         server::run_server(analysis, vfs, build_queue);
     }
 }
+
+fn version() -> &'static str {
+    concat!(env!("CARGO_PKG_VERSION"), include_str!(concat!(env!("OUT_DIR"), "/commit-info.txt")))
```

Comment:
@nrc the argument for `include_str!` must be a string literal. So we cannot do it without const fn.